components/raftstore/src/store/peer_storage.rs

```diff
@@ -1335,6 +1346,27 @@ where
         self.check_applying_snap() != CheckApplyingSnapStatus::Applying
     }
 
+    /// Cancel generating snapshot, return true if there is such a job.
+    pub fn cancel_generating_snap(&mut self, compact_to: Option<u64>) -> bool {
```

Comment:
Seems the return value is useless. Also if the `canceled` is true, don't need to check `index` again.